nxdialog/nxdialog

```diff
@@ -0,0 +1,369 @@
+  dlg.set_title(title)
+  dlg.set_markup(text)
+  dlg.add_button(DISCONNECT_TEXT, DISCONNECT)
+  dlg.add_button(TERMINATE_TEXT, TERMINATE)
+  dlg.add_button(CANCEL_TEXT, gtk.RESPONSE_CANCEL)
+
+  res = dlg.run()
+
+  if res in (DISCONNECT, TERMINATE):
+    return res
+
+  # Everything else is cancel
+  return None
+
+
+def ShowYesNoBox(title, text):
+  """ Shows a message box with answers yes and no.
+
+  @type title: str
+  @param title: Message box title
+  @type text: str
+  @param text: Message box text
+  @return: Choosen action
+
+  """
+  dlg = gtk.MessageDialog(type=gtk.MESSAGE_QUESTION, flags=gtk.DIALOG_MODAL)
+  dlg.set_title(title)
+  dlg.set_markup(text)
+  dlg.add_button(YES_TEXT, TERMINATE)
+  dlg.add_button(NO_TEXT, gtk.RESPONSE_CANCEL)
+
+  res = dlg.run()
+
+  if res == TERMINATE:
+    return res
+
+  # Everything else is cancel
+  return None
+
+
+def HandleSessionAction(agentpid, action):
+  """ Execute session action choosen by user.
+
+  @type agentpid: int
+  @param agentpid: Nxagent process id as passed by command line
+  @type action: int or None
+  @param action: Choosen action
+
+  """
+
+  if action == DISCONNECT:
+    print "Disconnecting from session, sending SIGHUP to %s" % (agentpid)
+    os.kill(agentpid, signal.SIGHUP)
+
+  elif action == TERMINATE:
+    print "Terminating session, sending SIGTERM to process %s" % (agentpid)
+    os.kill(agentpid, signal.SIGTERM)
+
+  elif action is None:
+    pass
+
+  else:
+    raise NotImplementedError()
+
+
+def ShowSimpleMessageBox(icon, title, text):
+  """ Shows a simple message box.
+
+  @type icon: QMessageBox.Icon
+  @param icon: Icon for message box
+  @type title: str
+  @param title: Message box title
+  @type text: str
+  @param text: Message box text
+
+  """
+  dlg = gtk.MessageDialog(type=icon, flags=gtk.DIALOG_MODAL,
+                          buttons=gtk.BUTTONS_OK)
+  dlg.set_title(title)
+  dlg.set_markup(text)
+  dlg.run()
+
+
+class NxDialogProgram(object):
+  """ the main program """
+  def __init__(self):
+    self.args = None
+    self.options = None
+
+  def Main(self):
+    """ let's do something """
+    try:
+      (self.options, self.args) = self.ParseArgs()
+
+      self.Run()
+
+    except (SystemExit, KeyboardInterrupt):
+      raise
+
+    except Exception, e:
+      sys.stderr.write("Caught exception: %s\n" % (e))
+      sys.exit(EXIT_FAILURE)
+
+  def ParseArgs(self):
+    """ init parser """
+    parser = optparse.OptionParser(option_list=self.BuildOptions(),
+                                   formatter=optparse.TitledHelpFormatter())
+    return parser.parse_args()
+
+  def BuildOptions(self):
+    """ build options for the parser """
+    return [
+        # nxagent 3.5.99.18 only used yesno, ok, pulldown and yesnosuspend
+        # yesno dialogs will always kill the session id "yes" is selected
+        optparse.make_option("--dialog", type="string", dest="dialog_type",
```

Comment:
@uli42 why not [argparse](https://docs.python.org/dev/library/argparse.html?highlight=s#upgrading-optparse-code)?

Reply:
because I have just adapted the original code from neatx which used optparse.

Reply:
converted to argparse